docs/system_overview/configuration.md

```diff
@@ -14,254 +14,682 @@ This scrambled password file is saved after an `iinit` is run. If this file does
+    "irods_ssl_dh_params_file": "",
+
+    // (Optional)
+    // Defines the size of the buffer used for sending and receiving bytes across the network
+    // during parallel transfer. Clients and servers within the same zone must use the same value.
+    "irods_transfer_buffer_size_for_parallel_transfer_in_megabytes": 4,
+
+    // The username within iRODS for this account.
+    "irods_user_name": "",
+
+    // The name of the iRODS Zone.
+    "irods_zone_name": "",
+
+    // (Optional)
+    // Defines the filename of the JSON schema used for validation when not specified explicitly.
+    // Only applies to service account environment files.
+    "schema_name": "service_account_environment",
+
+    // (Optional)
+    // Defines the version of the schema used for validation.
+    // Only applies to service account environment files.
+    "schema_version": "v4"
+}
+```
 
-To use an environment file other than `~/.irods/irods_environment.json`, set `IRODS_ENVIRONMENT_FILE` to load from a different location:
+A client environment file might look like this.
 
+```json
+{
+    "irods_host": "example.org",
+    "irods_port": 1247,
+    "irods_user_name": "alice",
+    "irods_zone_name": "tempZone"
+}
 ```
+
+And here is service account environment file defining the minimum settings needed to manage an iRODS server.
+
+```json
+{
+    "irods_client_server_negotiation": "request_server_negotiation",
+    "irods_client_server_policy": "CS_NEG_REFUSE",
+    "irods_connection_pool_refresh_time_in_seconds": 300,
+    "irods_cwd": "/tempZone/home/rods",
+    "irods_default_hash_scheme": "SHA256",
+    "irods_default_number_of_transfer_threads": 4,
+    "irods_default_resource": "demoResc",
+    "irods_encryption_algorithm": "AES-256-CBC",
+    "irods_encryption_key_size": 32,
+    "irods_encryption_num_hash_rounds": 16,
+    "irods_encryption_salt_size": 8,
+    "irods_home": "/tempZone/home/rods",
+    "irods_host": "example.org",
+    "irods_match_hash_policy": "compatible",
+    "irods_maximum_size_for_single_buffer_in_megabytes": 32,
+    "irods_port": 1247,
+    "irods_server_control_plane_encryption_algorithm": "AES-256-CBC",
+    "irods_server_control_plane_encryption_num_hash_rounds": 16,
+    "irods_server_control_plane_key": "32_byte_server_control_plane_key",
+    "irods_server_control_plane_port": 1248,
+    "irods_transfer_buffer_size_for_parallel_transfer_in_megabytes": 4,
+    "irods_user_name": "rods",
+    "irods_zone_name": "tempZone",
+    "schema_name": "service_account_environment",
+    "schema_version": "v4"
+}
+```
+
+To use an environment file other than `~/.irods/irods_environment.json`, set `IRODS_ENVIRONMENT_FILE` to load from a different location:
+
+```bash
 export IRODS_ENVIRONMENT_FILE=/full/path/to/different.json
 ```
 
 Other individual environment variables can be set by using the UPPERCASE versions of the variables named above, for example:
 
-```
+```bash
 export IRODS_LOG_LEVEL=7
```

Comment:
This option is deprecated, so maybe we shouldn't use it as the example anymore.

Reply:
that seems reasonable... what is a good replacement.... most-often overridden besides log level?

Reply:
Mmmm maybe `irods_cwd`/`IRODS_CWD`?

We don't have to replace this, just seemed like an accident to make it the example when we've marked it as deprecated. I feel like the environment variable override feature should be removed, personally, but maybe there's a use case for it that I'm not thinking of :)

Reply:
```bash
export IRODS_CWD='/tempZone/home/public'
```
That seems to work

Reply:
it's definitely a use case - people love some environment variables in their docker containers

Reply:
I was wondering if we should be advertising `IRODS_LOG_LEVEL` at all, but then I remembered we still need it for some tests because we haven't fully removed everything related to the old logging code.

For that reason, we do need to at least document it for ourselves.

Reply:
updated.